File: web/back/python-api/src/database/database_service.py
"""
Datei: database_service.py

Zweck:
Diese Datei prüft, ob PostgreSQL erreichbar ist.
"""

import psycopg

from src.config.settings import DATABASE_URL


def check_database_connection() -> dict:
    try:
        with psycopg.connect(DATABASE_URL, connect_timeout=2) as connection:
            with connection.cursor() as cursor:
                cursor.execute("SELECT 1")

        return {
            "connected": True,
            "message": "PostgreSQL ist erreichbar.",
        }
    except Exception as error:
        return {
            "connected": False,
            "message": str(error),
        }

File: web/back/java-api/src/main/java/com/example/javaapi/database/DatabaseService.java
package com.example.javaapi.database;

import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.stereotype.Service;

import java.util.Map;

/**
 * Datei: DatabaseService.java
 *
 * Zweck:
 * Diese Klasse prüft die PostgreSQL-Verbindung.
 */
@Service
public class DatabaseService {

    private final JdbcTemplate jdbcTemplate;

    public DatabaseService(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    public Map<String, Object> checkConnection() {
        try {
            jdbcTemplate.queryForObject("SELECT 1", Integer.class);

            return Map.of(
                    "connected", true,
                    "message", "PostgreSQL ist erreichbar."
            );
        } catch (Exception error) {
            return Map.of(
                    "connected", false,
                    "message", error.getMessage()
            );
        }
    }
}